zerver/migrations/0143_reactions_realm_emoji_name_to_id.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+import ujson
+
+from collections import defaultdict
+from django.conf import settings
+from django.db import migrations, models
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from typing import Any, Dict
+
+def realm_emoji_name_to_id(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Reaction = apps.get_model('zerver', 'Reaction')
+    RealmEmoji = apps.get_model('zerver', 'RealmEmoji')
+    realm_emojis_by_realm = defaultdict(dict)   # type: Dict[str, Dict[str, Any]]
+    for realm_emoji in RealmEmoji.objects.all():
+        realm_emojis_by_realm[realm_emoji.realm.name] = {
```

Comment:
I would use `realm_emoji.realm_id` as the key in this dict, not the `realm.name`; it's just better practice to use primary keys for this sort of thing, but I think it also saves doing a database query on every run through these loops.

Reply:
Agree, done! But didn't get how would it save database query?